test/domain_test.py

```diff
@@ -107,7 +106,7 @@ def test__pod_spec_is_generated():
 
         # Assertions
         assert isinstance(juju_pod_spec, domain.PrometheusJujuPodSpec)
-        assert juju_pod_spec.to_dict() == {'containers': [{
+        self.assertEqual(juju_pod_spec.to_dict(), {'containers': [{
```

Comment:
Why revert to this style?

Reply:
This allows us to have some more flexibility when troubleshooting the tests. For example, if `self.assertEqual` raises an exception - I have an option to use `self.maxDiff = None` to view the full and detailed diff, while I can't do this using regular `assert`.

Reply:
You can achieve that by adding `-vv` to the command actually. For example, with this test:

```
pytest --capture=no -vv test/domain_test.py::BuildJujuPodSpecTest::test__pod_spec_is_generated
```